script/script/views.py

```diff
@@ -0,0 +1,89 @@
+    ]
+    serializer_class = CountySerializer
+    # currently POST is allowed just for debugging
+    # http_method_names = ['get',]
+
+
+class EnergyViewSet(viewsets.ModelViewSet):
+    queryset = Energy.objects.all()
+    permission_classes = [
+        permissions.AllowAny,
+    ]
+    serializer_class = EnergySerializer
+
+
+class ChargingStationViewSet(viewsets.ModelViewSet):
+    queryset = ChargingStation.objects.all()
+    permission_classes = [
+        permissions.AllowAny,
+    ]
+    serializer_class = ChargingStationSerializer
+
+
+class EnergyView(mixins.RetrieveModelMixin,
+                        mixins.UpdateModelMixin,
+                        mixins.CreateModelMixin,
+                        mixins.DestroyModelMixin,
+                        generics.GenericAPIView):
+    queryset = Energy.objects.all()
+    permission_classes = [
+        permissions.AllowAny,
+    ]
+    serializer_class = EnergySerializer
+
+    # [TODO] get & validate the params in an additional func
+
+    def get(self, request, *args, **kwargs):
+        county_name = request.query_params.get('county', None)
+        county = County.objects.get(name=county_name)
+        year = request.query_params.get('year', None)
+        month = request.query_params.get('month', None)
+        res = Energy.objects.get(county=county, year=year, month=month)
+        serializer = self.serializer_class(res)
+        return Response(serializer.data)
+
+    def put(self, request, *args, **kwargs):
+        county_name = request.query_params.get('county', None)
+        county = County.objects.get(name=county_name)
+        year = request.query_params.get('year', None)
+        month = request.query_params.get('month', None)
+        energy = request.query_params.get('energy', None)
+        res = Energy.objects.get(county=county, year=year, month=month)
+        res.energy = energy
+        res.save()
+        serializer = self.serializer_class(res)
+        return Response(serializer.data)
+
+    def post(self, request, *args, **kwargs):
+        county_name = request.query_params.get('county', None)
+        county = County.objects.get(name=county_name)
+        year = request.query_params.get('year', None)
+        month = request.query_params.get('month', None)
+        energy = request.query_params.get('energy', None)
+        res = Energy.objects.create(county=county, year=year, month=month, energy=energy)
+        serializer = self.serializer_class(res)
+        return Response(serializer.data)
+
+    def delete(self, request, *args, **kwargs):
+        county_name = request.query_params.get('county', None)
+        county = County.objects.get(name=county_name)
+        year = request.query_params.get('year', None)
+        month = request.query_params.get('month', None)
+        res = Energy.objects.get(county=county, year=year, month=month)
+        res.delete()
+        serializer = self.serializer_class(res)
+        return Response(serializer.data)
```

Comment:
Thanks for your comments. This is my first time using DRF, and I got very confused about the design. I wrote two versions of the apis:
1. such as `/api/county/<county_name>/year/<year>`
2. such as `/api/county?county=<county_name>&year=<year>`
I actually prefer the former although the uri could be really long if there are many params in the composite unique keys. Then, the complete apis could be like:
```
GET: /api/county/<county_name>/year/<year>
PUT: /api/county/update
POST: /api/county/create
DELETE: /api/county/delete
``` 
What do you think of these apis?

Reply:
The GET endpoint is a little bit weird, so how about `/api/county/retrieve/name/<county_name>/year/<year>`?

Reply:
Thanks for the responses.

So for the GET request I think this would be a better approach:
```
GET: api/v1/county/{id}/?year=1985&year=1986&year.....

# This allows our backend to something like:

years = self.request.GET.getlist("year") 
```
[Reference](https://docs.djangoproject.com/en/2.2/ref/request-response/#django.http.QueryDict.getlist)

As far as the PUT, POST and DELETE, they don't need to be named anything, the VERB on the request will tell the api (DRF) the operation to do. All you need is supply the appropriate body and id. E.g:
```
PUT: api/v1/county/{id}/
# supply the appropriate json body to perform the update
{...}

POST: api/v1/county/
# supply the appropriate json body to perform the create
{...}

DELETE: api/v1/county/{id}/
# doesn't require anything else
```

So basically, by default, DRF's router creates detailed urls at /resource/{pk} and you GET, PUT, POST and DELETE them to retrieve, update, create and delete respectively.

Reply:
Thanks for your advice.
For the idea of filtering by list in GET, it is really helpful but I am going to do that in a new PR.